### PR TITLE
Middleware order

### LINKED ISSUE #15353: Include some common optional middleware in the order section
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/middleware/?view=aspnetcore-3.0#order

It would be good to go through our list of provided middleware and specify on the order section where they should go relative to other middleware.

See this issue as a reference for a case where there's a confusion in the middleware order:
https://github.com/aspnet/AspNetCore/issues/16584

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9898482c-fa54-7ce2-1716-d14fbe03990d
* Version Independent ID: 3926c686-0ccc-f9fe-ca52-0134612a1623
* Content: [ASP.NET Core Middleware](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/middleware/?view=aspnetcore-3.0#order)
* Content Source: [aspnetcore/fundamentals/middleware/index.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/fundamentals/middleware/index.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #15353

[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/middleware/index?view=aspnetcore-3.0&branch=pr-en-us-15542#middleware-order)

cc @serpent5

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@javiercn any middleware you'd like to add? Changed H2 to `Middleware order` for more SEO juice.

File: aspnetcore/fundamentals/middleware/index/snapshot/Startup22.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                    Configuration.GetConnectionString("DefaultConnection")));
+            services.AddDefaultIdentity<IdentityUser>()
+                .AddDefaultUI(UIFramework.Bootstrap4)
+                .AddEntityFrameworkStores<ApplicationDbContext>();
+
+            services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_2);
+        }
+
+#region snippet
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
```
Comment (guardrex) [inline aspnetcore/fundamentals/middleware/index/snapshot/Startup22.cs]: We usually inject IWebHostEnvironment

Comment (guardrex) [inline aspnetcore/fundamentals/middleware/index/snapshot/Startup22.cs]: Nevermind ... thought I was looking at 3.0. 🤦‍♂

**Comment by guardrex** [comment]
👍 :shipit:

**Comment by Rick-Anderson** [comment]
@javiercn  quick review.

**Comment by Rick-Anderson** [comment]
@javiercn quick review please or suggest another reviewer.

**Pull request merged**